org-gtd-files.el

```diff
@@ -44,31 +45,35 @@ This is the inbox. Everything goes in here when you capture it.
 (defconst org-gtd-default-file-name "org-gtd-tasks")
 
 ;;;###autoload
-(defun org-gtd-inbox-path ()
-  "Return the full path to the inbox file."
-  (org-gtd--path org-gtd-inbox))
+(defalias 'org-gtd-inbox-path 'org-gtd--ensure-inbox
```

Comment:
This is the last thing in this pull request that doesn't make a lot of sense from a code perspective. Either a function is private or it is public.

If I understand this correctly, there's an additional refactor because some code looked alike, but:

- nothing needs the path to the default gtd-file, if I understand correctly: the code makes abstraction of the file and only creates it if we need a target for refiling purposes from the inbox processing
- we do need the inbox path for the capture and refiling
- we use similar code to initialize both, because the code thinks about them both as an abstraction when it comes to making sure they exist.

So I think there's been too much logic refactored together, maybe?

I think something like this would make more sense -- example just for the inbox path. Basically, making sure the file exists is one thing, and returning the path is another thing. This allows us to have the function we want public (here's the path to a file that exists) and have the function we want private (create the file without bothering the user).

What do you think?

```elisp
(defun org-gtd-inbox-path ()
    "Return the full path to the inbox file."
    (let ((path (org-gtd--path org-gtd-inbox)))
      (org-gtd--ensure-file-exists org-gtd-inbox org-gtd-inbox-template)
      path))

(defun org-gtd--ensure-file-exists (path initial-contents)
  "Create the file at PATH with INITIAL-CONTENTS if it does not exist."
  (unless (f-exists-p path)
    (with-current-buffer (find-file-noselect path)
      (insert initial-contents)
      (org-mode-restart)
      (basic-save-buffer))))
```